### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by mentaljam**
These changes concern the optimizing of strings operations.

### CODE REVIEW

**Comment by jpetrell** [comment]
LGTM. Code quality-wise good changes, though performance overheads being saved here are neglible when considering where the total application launch time goes to. Missing change log tag and MER bug to refer to.

**Comment by martyone** [comment]
Readability suffers. I would prefer utilising QT_USE_QSTRINGBUILDER and QT_RESTRICTED_CAST_FROM_ASCII instead.

**Comment by mentaljam** [comment]
The PR was updated:
- All the commits was merged into one
- The QT_USE_QSTRINGBUILDER macro was used instead of the QString::append() method
- The commit message was fixed

**Comment by martyone** [comment]
I would omit the explicit use of QChar constructor so that it is consistent with the implicit QString contructor calls.

**Comment by mentaljam** [comment]
Not sure I understand you correctly, but I'll try to answer.

Due to Qt's sources the `QString::startsWith(QChar ...)` is equal to `QString[0] == QChar` and comparison with a `QString` leads to more complex calls.

Using the `QString::lastIndexOf(...)` in `SailfishAppPriv::dataDir()` looks uncleared but it also leads to fewer calls.

I understand that all these changes are like a drop in the sea in comparison with the total application launch time but a drop to a drop at some time become a river.

**Comment by martyone** [comment]
I mean instead of `argv0.startsWith(QChar('/'))` you can use just `argv0.startsWith('/')` - `QChar` constructor will be called implicitly. Similar for the `lastIndexOf` call.

File: src/sailfishapp_priv.cpp [hunk 1]
```diff
@@ -79,9 +79,9 @@ QString dataDir()
     QFileInfo exe = QFileInfo(applicationPath());
 
     // "/usr/bin/<appname>" --> "/usr/share/<appname>/"
-    return QDir::cleanPath(QString("%1/%2")
-        .arg(exe.absoluteDir().filePath("../share"))
-        .arg(appName()));
+    QString path = exe.absolutePath();
+    int i = path.lastIndexOf(QChar('/')) + 1;
+    return path.replace(i, path.length() - i, "share/") + appName();
```
Comment (pvuorela) [inline src/sailfishapp_priv.cpp]: Old issue, but the above comment has extra / that is not getting included in the result.

**Comment by pvuorela** [comment]
Let's just have this. Though doesn't really matter anywhere how much this is optimized. Primary focus should be readability.

**Pull request merged**